src/Windows/R/Editor/Impl/Settings/REditorSettings.cs

```diff
@@ -24,15 +26,28 @@ public sealed class REditorSettings: EditorSettings, IWritableREditorSettings, I
 
         private RFormatOptions _formatOptions = new RFormatOptions();
 
-        public REditorSettings(ICoreShell coreShell): base(coreShell, RContentTypeDefinition.ContentType) { }
-        public REditorSettings(IEditorSettingsStorage storage) : base(storage) { }
+        public REditorSettings(ICoreShell coreShell) : base(coreShell, RContentTypeDefinition.ContentType) {
+            CreateLintOptions();
+        }
+
+        public REditorSettings(IEditorSettingsStorage storage) : base(storage) {
+            CreateLintOptions();
+        }
 
         public override void ResetSettings() {
             _formatOptions = new RFormatOptions();
-            LintOptions = new LintOptions();
+            CreateLintOptions();
             base.ResetSettings();
         }
 
+        private void CreateLintOptions() {
+            LintOptions = new LintOptions();
+            LintOptions.PropertyChanged += _OnLintOptionsPropertyChanged;
```

Comment:
Back to storages (I'm still trying to understand the whole picture). `IRSettings` implements `INotifyPropertyChanged`, which is used to update `RSession` code page and CRAN mirror, but isn't used to trigger property saving. `DialogPage` doesn't save anything directly, it saves values through `IRSettings.SaveSettingsAsync`, so in fact we have two different places to save values: `ISettingsStorage` (implemented by `VsSettingsStorage`) and `IWritableEditorSettingsStorage` (implemented by `LanguageSettingsStorage`).

So where do you want to store `LintOptions` - in `ISettingsStorage` or in `IWritableEditorSettingsStorage`?